driver/src/main/java/org/neo4j/driver/internal/util/ServerVersion.java

```diff
@@ -28,28 +29,35 @@
 
 public class ServerVersion
 {
-    public static final ServerVersion v3_5_0 = new ServerVersion( 3, 5, 0 );
-    public static final ServerVersion v3_4_0 = new ServerVersion( 3, 4, 0 );
-    public static final ServerVersion v3_2_0 = new ServerVersion( 3, 2, 0 );
-    public static final ServerVersion v3_1_0 = new ServerVersion( 3, 1, 0 );
-    public static final ServerVersion v3_0_0 = new ServerVersion( 3, 0, 0 );
-    public static final ServerVersion vInDev = new ServerVersion( Integer.MAX_VALUE, Integer.MAX_VALUE, Integer.MAX_VALUE );
+    public static final ServerVersion v3_5_0 = new ServerVersion( "Neo4j", 3, 5, 0 );
+    public static final ServerVersion v3_4_0 = new ServerVersion( "Neo4j", 3, 4, 0 );
+    public static final ServerVersion v3_2_0 = new ServerVersion( "Neo4j", 3, 2, 0 );
+    public static final ServerVersion v3_1_0 = new ServerVersion( "Neo4j", 3, 1, 0 );
+    public static final ServerVersion v3_0_0 = new ServerVersion( "Neo4j", 3, 0, 0 );
+    public static final ServerVersion vInDev = new ServerVersion( "Neo4j", Integer.MAX_VALUE, Integer.MAX_VALUE, Integer.MAX_VALUE );
 
     private static final String NEO4J_IN_DEV_VERSION_STRING = "Neo4j/dev";
     private static final Pattern PATTERN =
-            Pattern.compile( "(Neo4j/)?(\\d+)\\.(\\d+)(?:\\.)?(\\d*)(\\.|-|\\+)?([0-9A-Za-z-.]*)?" );
+            Pattern.compile( "([^/]*)/(\\d+)\\.(\\d+)(?:\\.)?(\\d*)(\\.|-|\\+)?([0-9A-Za-z-.]*)?" );
```

Comment:
This will match strings like "/3.5.0" where the product part is absent. It is okay?